sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -715,6 +715,14 @@ object RapidsBufferCatalog extends Logging with Arm {
     deviceStorage = rdms
   }
 
+  // For testing
+  def setCatalog(catalog: RapidsBufferCatalog): Unit = {
```

Comment:
This needs to be synchronized. Can we add scala docs to it so the "For testing" actually would show up in the IDE when someone hovered over the API. 

Also the lazy initialization of `_singleton` in `def singleton` assumes no race conditions. We really should be rechecking for `_singleton == null` after grabbing the lock.

Reply:
yikes, that was not intended

Reply:
fixed.